frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1412,13 +1413,16 @@ impl<T: Config> Pallet<T> {
 	/// Extracted for easier weight calculation.
 	fn create_snapshot_external(
 	) -> Result<(Vec<T::AccountId>, Vec<VoterOf<T>>, u32), ElectionError<T>> {
-		let target_limit = T::MaxElectableTargets::get().saturated_into::<usize>();
-		let voter_limit = T::MaxElectingVoters::get().saturated_into::<usize>();
+		let election_bounds = T::ElectionBounds::get();
+		let target_limit =
+			election_bounds.targets.count.unwrap_or(u32::MAX).saturated_into::<usize>();
+		let voter_limit =
+			election_bounds.voters.count.unwrap_or(u32::MAX).saturated_into::<usize>();
 
-		let targets = T::DataProvider::electable_targets(Some(target_limit))
+		let targets = T::DataProvider::electable_targets(election_bounds.targets)
 			.map_err(ElectionError::DataProvider)?;
 
-		let voters = T::DataProvider::electing_voters(Some(voter_limit))
+		let voters = T::DataProvider::electing_voters(election_bounds.voters)
 			.map_err(ElectionError::DataProvider)?;
 
 		if targets.len() > target_limit || voters.len() > voter_limit {
```

Comment:
We can just get rid of this. It is double checking some bounds that are already "guaranteed" to be met. If we are to check, we should also check the byte size (if possible to do cheaply).

Reply:
Added voter size checks here too.

Reply:
Your point about this pallet not dealing with byte size is correct, but in extension it would also not have to deal with `length` either. 

In principle, this pallet only knows that there is an `election_bounds`, which has `voters` and `targets`, and once passed to the `T::DataProvider`, they need to be `respected`.

If you were to do this pedantically, you would make the fields of `ElectionProviderBounds` private to hide this actually. 

I would formulate this then as:

```
let election_bounds = T::ElectionBounds::get();

let targets = T::DataProvider::electable_targets(election_bounds.targets)
	.and_then(|t| election_bounds.targets.ensure_respected(&v))
	.map_err(ElectionError::DataProvider)?;

let voters = T::DataProvider::electing_voters(election_bounds.voters)
	.and_then(|v| election_bounds.voters.ensure_respected(&v))
	.map_err(ElectionError::DataProvider)?;
```

and then inside `ensure_respected` implementation you can check length, and check byte-size again as well.